src/test/abi_component/widget.cpp

```diff
@@ -0,0 +1,145 @@
+#include "iwidget.h"
+#include <pal_error.h>
+#include <string_view>
+
+struct widget : iwidget
+{
+    int32_t XLANG_CALL QueryInterface(xlang_guid const& id, void** object) noexcept final
```

Comment:
why is there a mix of noexcept and XLANG_NOEXCEPT?

Reply:
again copy/paste from existing PAL files. I'll let @DefaultRyan comment, but I suspect these should all be noexcept. XLANG_NOEXCEPT appears to be there to enable pal.h to be compiled by a C compiler.